sycl/source/detail/config.hpp

```diff
@@ -467,6 +472,125 @@ template <> class SYCLConfig<SYCL_CACHE_DIR> {
   }
 };
 
+template <> class SYCLConfig<SYCL_REDUCTION_PREFERRED_WORKGROUP_SIZE> {
+  using BaseT = SYCLConfigBase<SYCL_REDUCTION_PREFERRED_WORKGROUP_SIZE>;
+
+  struct ParsedValue {
+    size_t CPU = 0;
+    size_t GPU = 0;
+    size_t Accelerator = 0;
+  };
+
+public:
+  static size_t get(info::device_type DeviceType) {
+    ParsedValue Value = getCachedValue();
+    return getRefByDeviceType(Value, DeviceType);
+  }
+
+  static void reset() { (void)getCachedValue(/*ResetCache=*/true); }
+
+  static const char *getName() { return BaseT::MConfigName; }
+
+private:
+  static size_t &getRefByDeviceType(ParsedValue &Value,
+                                    info::device_type DeviceType) {
+    switch (DeviceType) {
+    case info::device_type::cpu:
+      return Value.CPU;
+    case info::device_type::gpu:
+      return Value.GPU;
+    case info::device_type::accelerator:
+      return Value.Accelerator;
+    default:
+      throw INVALID_CONFIG_EXCEPTION(
+          BaseT, "Device types must be \"cpu\", \"gpu\", \"acc\", or \"*\".");
```

Comment:
We aren't handling `*` here, are we?

Reply:
We are not, but the message is most relevant for users rather than internal. I.e. if a user gives a wrong device type we want to also include * in the message to make sure they know it's an option. If get uses an invalid value for this then this doesn't help the user much anyway.

I could try and move it out of this function, but it will need a bit of refactoring.